qiskit_machine_learning/circuit/library/raw_feature_vector.py

```diff
@@ -112,11 +117,13 @@ def num_qubits(self, num_qubits: int) -> None:
         Args:
             The new number of qubits.
         """
-        if self._num_qubits != num_qubits:
+        if self.num_qubits != num_qubits:
             # invalidate the circuit
             self._invalidate()
-            self._num_qubits = num_qubits
-            self.add_register(QuantumRegister(self.num_qubits, "q"))
+            if num_qubits:
+                self.add_register(QuantumRegister(num_qubits, "q"))
```

Comment:
You should probably reset the registers, otherwise calling the setter twice will add too many registers (and qubits!) and complain that "q" already exists:
```python
>>> from qiskit_machine_learning.circuit.library import RawFeatureVector
>>> r = RawFeatureVector(2)
>>> r.num_qubits = 1
>>> r.draw()
     ┌────────────────────────────────────┐
q_0: ┤ Parameterizedinitialize(x[0],x[1]) ├
     └────────────────────────────────────┘
>>> r.num_qubits = 3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jul/Qiskit/qiskit-machine-learning/qiskit_machine_learning/circuit/library/raw_feature_vector.py", line 124, in num_qubits
    self.add_register(QuantumRegister(num_qubits, "q"))
  File "/Users/jul/Qiskit/qiskit-terra/qiskit/circuit/quantumcircuit.py", line 1179, in add_register
    raise CircuitError('register name "%s" already exists' % register.name)
qiskit.circuit.exceptions.CircuitError: 'register name "q" already exists'
```

Reply:
Yes!!! I actually ran into this situation with the original implementation but forgot about it. Thanks!